ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java

```diff
@@ -1011,6 +1109,8 @@ public void onLayoutChange(
       return;
     }
 
+    updateScrollPositionForMaintainVisibleContentPosition();
```

Comment:
Yes, there are cases when removing views where even after adjustments we want to do the other adjustment (in the following lines) to make sure scroll position is not outside of the content view. Since we call `getScrollY` again it should already use the updated value from `updateScrollPositionForMaintainVisibleContentPosition`.